build-support/bin/ci.py

```diff
@@ -218,9 +218,6 @@ def pants_command(
                         "--no-v1",
                         "--v2",
                         "--pants-config-files=pants.remote.ini",
-                        # We turn off speculation to reduce the risk of flakiness, where a test
-                        # passes either locally or remotely and fails in the other environment.
-                        "--process-execution-speculation-strategy=none",
```

Comment:
Is it still a risk that a test would fail with remote, but because it passes locally we don't end up running remote?

Even if so, probably worth using speculation for a) dogfooding and b) reduce the monetary cost of remote execution.

Reply:
> Is it still a risk that a test would fail with remote, but because it passes locally we don't end up running remote?

If we want to guarantee that remote is always passing, we'd want to swap the primary and second here (IE, run with "remote first").